### PR TITLE
Emad/[SPDBT-2961] controlling member auth save

### PR DESCRIPTION
**Pull request by esdd1995**
# Description

This PR includes the following proposed change(s):

- added auth save controller
- added manager for controlling member crc app auth save
- added repository for auth save controlling member crc app
- mappings for auth save controlling member crc app
- moved `LinkTeam` to `SharedRepositoryFuncs` and related changes.
- assigned `WorkerLicenceTypeCode` to controlling memeber crc enum option.

### CODE REVIEW

File: src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs [hunk 1]
```diff
@@ -62,7 +64,24 @@ public async Task<ControllingMemberCrcAppCommandResponse> Handle(ControllingMemb
...
-    private static void ValidateFilesForNewApp(ControllingMemberCrcAppSubmitRequestCommand cmd)
+    #endregion
+    public async Task<ControllingMemberCrcAppCommandResponse> Handle(ControllingMemberCrcUpsertCommand cmd, CancellationToken ct)
+    {
+        SaveControllingMemberCrcAppCmd saveCmd = _mapper.Map<SaveControllingMemberCrcAppCmd>(cmd.ControllingMemberCrcAppUpsertRequest);
+        
+        //TODO: find the purpose, add related enums if needed (ask peggy)
+        saveCmd.UploadedDocumentEnums = GetUploadedDocumentEnumsFromDocumentInfo((List<Document>?)cmd.ControllingMemberCrcAppUpsertRequest.DocumentInfos);
+
+        var response = await _controllingMemberCrcRepository.SaveControllingMenberCrcApplicationAsync(saveCmd, ct);
```
Comment (carolcarpenter) [inline src/Spd.Manager.Licence/ControllingMemberCrcAppManager.cs:75]: spelling 'Member'

File: src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs [hunk 2]
```diff
@@ -63,4 +64,48 @@ public async Task<ControllingMemberCrcApplicationCmdResp> CreateControllingMembe
...
+        {
+            app = _mapper.Map<spd_application>(cmd);
+            _context.AddTospd_applications(app);
+            var contact = _context.contacts.Where(l => l.contactid == cmd.ContactId).FirstOrDefault();
+            if (contact != null)
+            {
+                _context.SetLink(app, nameof(spd_application.spd_ApplicantId_contact), contact);
+            }
+        }
+        //TODO: what is the LinkServiceType??
```
Comment (peggy-quartech) [inline src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs:91]: It should be ControllingMemberCRC

File: src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs [hunk 3]
```diff
@@ -63,4 +64,48 @@ public async Task<ControllingMemberCrcApplicationCmdResp> CreateControllingMembe
...
+        SharedRepositoryFuncs.LinkServiceType(_context, cmd.WorkerLicenceTypeCode, app);
+   
+    
+        //TODO: what is the LinkTeam ??
+        await LinkTeam(DynamicsConstants.Licensing_Client_Service_Team_Guid, app, ct);
+        await _context.SaveChangesAsync();
+      
+
+        //TODO: do we have category codes in the crc controlling member process?
```
Comment (peggy-quartech) [inline src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs:100]: there is no categoryCode for controllingMember, if it has, your payload would include that.

File: src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs [hunk 4]
```diff
@@ -63,4 +64,48 @@ public async Task<ControllingMemberCrcApplicationCmdResp> CreateControllingMembe
...
+        //SharedRepositoryFuncs.ProcessCategories(_context, cmd.CategoryCodes, app);
+        //await _context.SaveChangesAsync();
+        return new ControllingMemberCrcApplicationCmdResp((Guid)app.spd_applicationid, (Guid)cmd.ContactId);
+    }
+    private async Task LinkTeam(string teamGuidStr, spd_application app, CancellationToken ct)
```
Comment (peggy-quartech) [inline src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs:105]: there are several place in the code use LinkTeam, LinkServiceType,
Could we move them to a shared place?

Comment (esdd1995) [reply src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs:105]: I agree with that, I'll check it

**Review by peggy-quartech — APPROVED** [review]

**Pull request merged**